typescript/sdk/src/metadata/agentConfig.ts

```diff
@@ -201,6 +255,13 @@ export const AgentChainMetadataSchema = ChainMetadataSchemaObject.merge(
       }
     }
 
+    // If the protocol type is Sealevel, require everything in AgentSealevelChainMetadataSchema
+    if (metadata.protocol === ProtocolType.Sealevel) {
+      if (!AgentSealevelChainMetadataSchema.safeParse(metadata).success) {
+        return false;
```

Comment:
nit: `return AgentSealevelChainMetadataSchema.safeParse(metadata).success`

Reply:
think it may be better keep as-is so that if anyone adds more checks after this line in the future, we don't return true earlier than we actually want to. Not sure it'd be obvious to someone of the sharp edge there. I assume that was the rationale for the cosmos parsing a little above this